lib/common/tests/xml/pcmk__xe_foreach_child_test.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2022 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <crm/common/unittest_internal.h>
+#include <crm/common/xml_internal.h>
+
+static int compare_name_handler(xmlNode *xml, void *userdata) {
+    function_called();
+    assert_string_equal((char *) userdata, crm_element_name(xml));
+    return pcmk_rc_ok;
+}
+
+const char *str1 =
+    "<xml>\n"
+    "  <level1>\n"
+    "  </level1>\n"
+    "  <level1>\n"
+    "  </level1>\n"
+    "  <level1>\n"
+    "  </level1>\n"
+    "</xml>";
```

Comment:
I just realized it would be a good idea to throw some comment and text nodes in the inputs to make sure they're ignored

Reply:
Good idea.